### PR TITLE
[#1493] DevOps Guide: move from Wiki to DG

### LINKED ISSUE #1493: Move DevOps guide in the wiki to the developer guide
The [DevOps guide](../wiki/DevOps-guide) in the wiki should be moved to a page of its own in the [RepoSense developer guide](https://reposense.org/dg/index.html).

This task is good for first timers to understand the workflow for contributing to RepoSense.

### PR DESCRIPTION
**Pull request by dcshzj**
Fixes #1493.

Reference: [Original wiki page](/reposense/RepoSense/wiki/DevOps-guide)

```
The DevOps guide was originally created in the wiki as a draft,
and it should be moved to the RepoSense developer guide for better
visibility.

Let's move the contents of that wiki page into a new page in the
developer guide.
```

### CODE REVIEW

**Review by Tejas2805 — CHANGES_REQUESTED** [review]
Just change all internal headings to sentence case to make it uniform with the rest of the docs and we should be good to go.

**Comment by damithc** [comment]
@dcshzj any updates? Let me when ready so that I can have a look too.

**Comment by dcshzj** [comment]
Apologies for the delay @Tejas2805, I have updated the headings.

**Comment by HCY123902** [comment]
Is the preview currently available? I cannot seem to open it

**Comment by dcshzj** [comment]
Looks like an issue with MarkBind v3, as we do not have a default layout file.

**Comment by HCY123902** [comment]
> Looks like an issue with MarkBind v3, as we do not have a default layout file.

This may be potentially related to #1557. We can start a separate issue on this

File: docs/dg/devOpsGuide.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+2. Surge.sh build preview (defined in [`surge.yml`](https://github.com/reposense/RepoSense/blob/master/.github/workflows/surge.yml)): Used for actually deploying the RepoSense report and documentation website to Surge.sh and updating the pull request checklist to link to the preview websites.
+
+The previews are recognised as GitHub deployments and are named `dashboard-$PRNUMBER` and `docs-$PRNUMBER`, where `$PRNUMBER` is the pull request number. Once a preview is ready, the reviewer would be able to click on "View deployment" or "Show environments" in the pull request to open the preview websites.
+
+Both the "Surge.sh pending build" and "Continuous Integration" workflows produce an artifact respectively, which is then downloaded by the "Surge.sh build preview" workflow to be deployed to Surge.sh. Due to limitations in GitHub Actions in determining the workflow execution order, the "Surge.sh pending build" workflow is assumed to complete before the "Continuous Integration" workflow is completed.
+
+This task is not performed on commits to the repository, as there is no need to do so.
```
Comment (HCY123902) [inline docs/dg/devOpsGuide.md:54]: Actually, what does this mean? Do you mean a direct commit to the upstream repository will not trigger the workflow?

Comment (dcshzj) [reply docs/dg/devOpsGuide.md:54]: Yep, once the PR is merged, it creates a commit into the master branch, but this workflow will not run during that commit.

**Comment by damithc** [comment]
> Looks like an issue with MarkBind v3, as we do not have a default layout file.

I've updated https://github.com/reposense/RepoSense/blob/master/.github/workflows/gh-pages.yml to use an older version of MarkBind, until we migrate to MarkBind v3 (which is not fully stable yet anyway).

**Comment by dcshzj** [comment]
@HCY123902 The documentation preview works now.

**Review by fzdy1914 — APPROVED** [review]
LGTM, a very nice document.

**Pull request merged**